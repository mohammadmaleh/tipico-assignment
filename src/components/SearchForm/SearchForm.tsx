import * as React from "react";
import { Input, Button } from "semantic-ui-react";
interface ISearchFormProps {
  onChangeSearchInput?: React.ChangeEventHandler<
    HTMLTextAreaElement | HTMLInputElement
  >;
  onSearchButtonClicked?:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
}
const styles = {
  container: {
    height: 70,
    display: "flex",
    justifyContent: "center",
    alignContent: "center"
  },
  searchInput: {
    margin: 10
  },
  submitButton: {
    margin: 10
  }
};
const SearchForm: React.FunctionComponent<ISearchFormProps> = props => {
  const { onChangeSearchInput, onSearchButtonClicked } = props;
  return (
    <div data-test="component-search-form" style={styles.container}>
      <Input
        onChange={onChangeSearchInput}
        style={styles.searchInput}
        data-test="search-input"
      />
      <Button
        onClick={onSearchButtonClicked}
        style={styles.submitButton}
        data-test="submit-button"
      >
        Search
      </Button>
    </div>
  );
};

export default SearchForm;
