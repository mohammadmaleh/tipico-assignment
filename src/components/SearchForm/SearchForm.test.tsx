import * as React from "react";
import { shallow } from "enzyme";
import SearchForm from "./SearchForm";
import { findByTestAttr } from "../../../test/testUtils";
const defaultProps = {};

const setup = (props = {}) => {
  const setupProps = { ...props };
  return shallow(<SearchForm {...setupProps} />);
};
describe("search form", () => {
  test("it renders without error", () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, "component-search-form");
    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
  test("renders search input", () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, "search-input");
    expect(component.length).toBe(1);
  });
  test("renders submit button", () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, "submit-button");
    expect(component.length).toBe(1);
  });
});
