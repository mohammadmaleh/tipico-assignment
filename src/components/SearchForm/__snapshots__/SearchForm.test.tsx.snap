// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search form it renders without error 1`] = `ShallowWrapper {}`;
